contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol

```diff
@@ -36,33 +36,43 @@ interface ILinkAvailable {
      we could save a fair amount of gas and re-write this upkeep for use with Automation v2.0+,
      which has significantly different trust assumptions
  */
-contract LinkAvailableBalanceMonitor is ConfirmedOwner, Pausable, KeeperCompatibleInterface {
+contract LinkAvailableBalanceMonitor is ConfirmedOwner, Pausable, AutomationCompatibleInterface {
   using EnumerableMap for EnumerableMap.AddressToUintMap;
 
   event FundsWithdrawn(uint256 amountWithdrawn, address payee);
   event TopUpSucceeded(address indexed topUpAddress);
   event TopUpBlocked(address indexed topUpAddress);
   event WatchlistUpdated();
+  event MaxPerformUpdated(uint256 oldMaxPerform, uint256 newMaxPerform);
+  event MaxCheckUpdated(uint256 oldMaxCheck, uint256 newMaxCheck);
 
   error InvalidWatchList();
   error DuplicateAddress(address duplicate);
 
-  uint256 public constant MAX_PERFORM = 5; // max number to addresses to top up in a single batch
-  uint256 public constant MAX_CHECK = 20; // max number of upkeeps to check (need to fit in 5M gas limit)
   IERC20 public immutable LINK_TOKEN;
 
   EnumerableMap.AddressToUintMap private s_watchList;
   uint256 private s_topUpAmount;
+  uint256 private s_maxPerform; // max number to addresses to top up in a single batch
+  uint256 private s_maxCheck; // max number of upkeeps to check (need to fit in 5M gas limit)
+  uint256 private s_minWaitPeriodSeconds;
```

Comment:
nit: this var will be used in future PR's, please don't mind it being unused.

Reply:
Since they are not constants anymore, their size matters a lot in the gas cost. I feel like all of these new uint256s could be a lot smaller and fit into a single slot https://dev.to/javier123454321/solidity-gas-optimizations-pt-3-packing-structs-23f4

Reply:
Reduced down to uint32 (minWait) and uint16 (maxCheck, maxPerform), should be enough for these parameters.